File: CG/CourceWork/CourceWork/IF_2D.cpp
#include "IF_2D.h"

Point2D::Point2D() {
	setX(0);
	setY(0);
}

Point2D::Point2D(double x, double y) {
	setX(x);
	setY(y);
}

double Point2D::getX() {
	return x;
}
double Point2D::getY() {
	return y;
}
void Point2D::setX(double x) {
	this->x = x;
}
void Point2D::setY(double y) {
	this->y = y;
}

IF_2D::IF_2D(double aX, double bX, double dX, double aY, double bY, double dY, double chance) {
	setaX(aX);
	setbX(bX);
	setdX(dX);
	setaY(aY);
	setbY(bY);
	setdY(dY);
	setChance(chance);
}
IF_2D::IF_2D() {
	setaX(0);
	setbX(0);
	setdX(0);
	setaY(0);
	setbY(0);
	setdY(0);
	setChance(0);
}
IF_2D::~IF_2D() {

}
double IF_2D::getaX() {
	return aX;
}

void IF_2D::setaX(double aX) {
	this->aX = aX;
}

double IF_2D::getbX() {
	return bX;
}

void IF_2D::setbX(double bX) {
	this->bX = bX;
}

double IF_2D::getdX() {
	return dX;
}

void IF_2D::setdX(double dX) {
	this->dX = dX;
}

double IF_2D::getaY() {
	return aY;
}

void IF_2D::setaY(double aY) {
	this->aY = aY;
}

double IF_2D::getbY() {
	return bY;
}

void IF_2D::setbY(double bY) {
	this->bY = bY;
}

double IF_2D::getdY() {
	return dY;
}

void IF_2D::setdY(double dY) {
	this->dY = dY;
}

double IF_2D::getChance() {
	return chance;
}

void IF_2D::setChance(double chance) {
	this->chance = chance;
}

Point2D IF_2D::calculate(Point2D point) {
	return Point2D(getaX()*point.getX() + getbX()*point.getY() + getdX(), getaY()*point.getX() + getbY()*point.getY() + getdY());
}

File: CG/CG_COURSEWORK/src/algorithm/IF_2D.java
package algorithm;

import javafx.geometry.Point2D;

/**
 * Created by Sergey on 24.01.2017.
 */
//Аффинное преобразование
public class IF_2D {
    //Значения для функции X'=A*X+B*Y+C
    double aX;
    double bX;
    double dX;
    //Значения для функции Y'=D*X+E*Y+F
    double aY;
    double bY;
    double dY;
    //Вероятность выбора преобразования
    double chance;

    //Конструктор
    public IF_2D(double aX, double bX, double dX, double aY, double bY, double dY, double chance) {
        setaX(aX);
        setbX(bX);
        setdX(dX);
        setaY(aY);
        setbY(bY);
        setdY(dY);
        setChance(chance);
    }

    //region Сеттеры и геттеры
    public double getaX() {
        return aX;
    }

    public void setaX(double aX) {
        this.aX = aX;
    }

    public double getbX() {
        return bX;
    }

    public void setbX(double bX) {
        this.bX = bX;
    }

    public double getdX() {
        return dX;
    }

    public void setdX(double dX) {
        this.dX = dX;
    }

    public double getaY() {
        return aY;
    }

    public void setaY(double aY) {
        this.aY = aY;
    }

    public double getbY() {
        return bY;
    }

    public void setbY(double bY) {
        this.bY = bY;
    }

    public double getdY() {
        return dY;
    }

    public void setdY(double dY) {
        this.dY = dY;
    }

    public double getChance() {
        return chance;
    }

    public void setChance(double chance) {
        this.chance = chance;
    }
    //endregion

    //Провести преобразование
    public Point2D calculate(Point2D point){
        return new Point2D(getaX()*point.getX()+getbX()*point.getY()+getdX(), getaY()*point.getX()+getbY()*point.getY()+getdY());
    }
}
